collection+xattrs.go

```diff
@@ -236,28 +244,102 @@ func (c *Collection) WriteUpdateWithXattr(
 func (c *Collection) DeleteWithXattr(
 	key string,
 	xattrKey string,
-) error {
+) (err error) {
 	traceEnter("DeleteWithXattr", "%q, %q", key, xattrKey)
-	err := c.withNewCas(func(txn *sql.Tx, newCas CAS) (*event, error) {
-		e := &event{
-			key:        key,
-			cas:        newCas,
-			isDeletion: true,
-		}
-		row := txn.QueryRow(`SELECT xattrs FROM documents WHERE collection=?1 AND key=?2`, c.id, key)
-		var rawXattrs []byte
-		err := scan(row, &rawXattrs)
-		if err != nil {
-			return nil, remapKeyError(err, key)
-		}
-		rawXattrs = removeXattrs(rawXattrs, xattrKey)
-		e.xattrs = rawXattrs
+	defer func() { traceExit("DeleteWithXattr", err, "ok") }()
+	return c._deleteBodyAndXattr(key, xattrKey, false)
+}
 
-		_, err = txn.Exec(`UPDATE documents SET value=null, xattrs=?1, cas=?2 WHERE collection=?3 AND key=?4`, rawXattrs, newCas, c.id, key)
-		return e, err
-	})
-	traceExit("DeleteWithXattr", err, "ok")
-	return err
+//////// XattrStore2 INTERFACE:
+
+// Creates a tombstone doc (no value) with an xattr.
+func (c *Collection) InsertXattr(
+	key string,
+	xattrKey string,
+	exp uint32,
+	cas uint64,
+	xv interface{},
+) (casOut uint64, err error) {
+	traceEnter("InsertXattr", "%q, %q, exp=%d, ...", key, xattrKey, exp)
+	defer func() { traceExit("InsertXattr", err, "0x%x", casOut) }()
+	return c.writeWithXattr(key, nil, xattrKey, payload{parsed: xv}, &cas, &exp, xInsertDoc+xInsertXattr)
+}
+
+// Creates a document, with an xattr, only if it doesn't already exist. (?)
```

Comment:
I was unsure about the semantics of some of the methods; those were my best guesses. I've since updated the descriptions in the interface declaration but forgot to update the copies here.